pkgs/by-name/gn/gnome-settings-daemon/package.nix

```diff
@@ -71,34 +85,41 @@ stdenv.mkDerivation (finalAttrs: {
     python3
   ];
 
-  buildInputs = [
-    gtk3
-    glib
-    gsettings-desktop-schemas
-    modemmanager
-    networkmanager
-    libnotify
-    libgnomekbd # for org.gnome.libgnomekbd.keyboard schema
-    gnome-desktop
-    libpulseaudio
-    alsa-lib
-    libcanberra-gtk3
-    upower
-    colord
-    libgweather
-    polkit
-    geocode-glib_2
-    geoclue2
-    systemd
-    libgudev
-    libwacom
-    gcr_4
-  ];
+  buildInputs =
+    [
+      gtk3
+      glib
+      gsettings-desktop-schemas
+      modemmanager
+      networkmanager
+      libnotify
+      libgnomekbd # for org.gnome.libgnomekbd.keyboard schema
+      gnome-desktop
+      libpulseaudio
+      alsa-lib
+      libcanberra-gtk3
+      upower
+      colord
+      libgweather
+      polkit
+      geocode-glib_2
+      geoclue2
+      libgudev
+      libwacom
+      gcr_4
+    ]
+    ++ lib.optionals withSystemd [
+      systemd
+    ];
 
-  mesonFlags = [
-    "-Dudev_dir=${placeholder "out"}/lib/udev"
-    "-Dgnome_session_ctl_path=${gnome-session-ctl}/libexec/gnome-session-ctl"
-  ];
+  mesonFlags =
+    [
+      "-Dudev_dir=${placeholder "out"}/lib/udev"
+      (lib.mesonBool "systemd" withSystemd)
+    ]
+    ++ lib.optionals withSystemd [
+      "-Dgnome_session_ctl_path=${gnome-session-ctl}/libexec/gnome-session-ctl"
```

Comment:
Looks like this is only used by the systemd service so it should be fine: https://github.com/NixOS/nixpkgs/blob/b69de56fac8c2b6f8fd27f2eca01dcda8e0a4221/pkgs/by-name/gn/gnome-settings-daemon/add-gnome-session-ctl-option.patch#L35